selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/ImportWizardFormTest.java

```diff
@@ -0,0 +1,614 @@
+  private static final String SECOND_BRANCH = "secondBranch";
+  private static final String THIRD_BRANCH = "thirdBranch";
+  private static final String FIRST_SUBMODULE_NAME = "Repo_For_Test";
+  private static final String SECOND_SUBMODULE_NAME = "testRepo";
+  public static final String DIRECTORY_NAME_1 = "my-lib";
+  public static final String DIRECTORY_NAME_2 = "my-lib/src/test";
+  private String currentProjectName;
+
+  @Inject private TestWorkspace ws;
+  @Inject private Ide ide;
+
+  @Inject
+  @Named("github.username")
+  private String gitHubUsername;
+
+  @Inject
+  @Named("github.password")
+  private String gitHubPassword;
+
+  @Inject
+  @Named("che.multiuser")
+  private boolean isMultiuser;
+
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private MachineTerminal terminal;
+  @Inject private Menu menu;
+  @Inject private ImportProjectFromLocation importProject;
+  @Inject private Preferences preferences;
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+  @Inject private SeleniumWebDriverHelper seleniumWebDriverHelper;
+  @Inject private AskDialog askDialog;
+  @Inject private GitHub gitHub;
+  @Inject private TestGitHubServiceClient gitHubClientService;
+  @Inject private Loader loader;
+  @Inject private Events events;
+  @Inject private TestUserPreferencesServiceClient testUserPreferencesServiceClient;
+  @Inject private Dashboard dashboard;
+  @Inject private KeycloakFederatedIdentitiesPage keycloakFederatedIdentitiesPage;
+  @Inject private TestGitHubRepository keepDirectoryRepo;
+  @Inject private TestGitHubRepository importBranchRepo;
+  @Inject private TestGitHubRepository submoduleRepo;
+  @Inject private Git git;
+  @Inject private Wizard projectWizard;
+  @Inject private CodenvyEditor editor;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+
+  @BeforeClass(groups = TestGroup.MULTIUSER)
+  @AfterClass(groups = TestGroup.MULTIUSER)
+  private void removeGitHubIdentity() {
+    dashboard.open(); // to login
+    keycloakFederatedIdentitiesPage.open();
+    keycloakFederatedIdentitiesPage.ensureGithubIdentityIsAbsent();
+    assertEquals(keycloakFederatedIdentitiesPage.getGitHubIdentityFieldValue(), "");
+  }
+
+  @BeforeClass
+  private void revokeGithubOauthToken() {
+    try {
+      gitHubClientService.deleteAllGrants(gitHubUsername, gitHubPassword);
+    } catch (Exception e) {
+      LOG.warn("There was an error of revoking the github oauth token.", e);
+    }
+  }
+
+  @BeforeClass
+  private void deletePrivateSshKey() throws Exception {
+    ide.open(ws);
+    projectExplorer.waitProjectExplorer();
+    terminal.waitTerminalTab();
+
+    openPreferencesVcsForm();
+
+    if (preferences.isSshKeyIsPresent(GITHUB_COM)) {
+      preferences.deleteSshKeyByHost(GITHUB_COM);
+    }
+
+    preferences.clickOnCloseBtn();
+  }
+
+  @Test
+  public void shouldLoginToGitHubAndImportProject() throws Exception {
+    currentProjectName = "AngularJS";
+
+    initRepoForFirstTest();
```

Comment:
Place test number into the name of method is not a good idea because it requires additional effort in case of test reordering. It's better to name methods by tests propose.